manifestoes/extensions.rst

```diff
@@ -0,0 +1,1675 @@
+    ``-Wmissing-kind-signatures``, ``-Wmissing-local-signatures``, ``-Wmissing-pattern-synonym-signatures``,
+    ``-Wmissing-role-annotations``, ``-Wmissing-signatures``
+
+  * ``Complete``: There are a number of places where programmers can omit parts of their program
+    and still get it to compile. The ``Complete`` option makes these into errors. All such features
+    are warnings by default (within the language editions, at least); ``Complete`` turns them into
+    errors. Users enabling this option will have to fix such errors long before deployment. (Other
+    warnings require fixing only for deployment, not during development.)
+
+    Includes the following: ``-Werror=incomplete-patterns``, ``-Werror=incomplete-record-selectors``,
+    ``-Werror=incomplete-record-updates``, ``-Werror=incomplete-uni-patterns``, ``-Werror=missing-fields``,
+    ``-Werror=missing-methods``
+
+  * ``T1``, ``T2``, and ``T3``. These options are meant to be analogous to optimization levels
+    ``O1``, ``O2``, and ``O3``, but for type inference. When a user specifies optimization, they
+    understand that the runtime behavior of their program may be less related to when they wrote and
+    thus harder to preduct, but presumably will be faster. Similarly, specifying a type inference
+    level beyond 0 means that GHC will work harder to accept their program, but the exact types inferred
+    become harder to predict.
+
+    * ``T1``: Type inference is expected to be stable, though it is harder (in corner cases) for users to guess
+      the inferred type. The type checker may also run forever, but it will never produce a program that
+      does so (unless the expressions in the program indeed loop).
+
+      Includes the following: ``FlexibleContexts``, ``FlexibleInstances``, ``LiberalTypeSynonyms``,
+      ``MultiParamTypeClasses``, ``TypeSynonymInstances``, ``UndecidableInstances``, ``UndecidableSuperClasses``
+
+    * ``T2``:  It is possible (though unlikely) that type inference
+      will change between major releases. This level enables *functional dependencies*, which
+      allows type inference to propagate information in new, sometimes unexpected ways.
+
+      Includes the following: ``FunctionalDependencies``, ``ImpredicativeTypes`` (Quick Look only),
+      ``NoMonoLocalBinds``, ``OVERLAP`` pragmas, ``TypeFamilyDependencies``
+
+    * ``T3``: The type inference engine is now allowed, in some scenarios, to make guesses
+      between two valid possibilities. These guesses might even influence runtime behavior.
+      Enabling this level of type inference should be done only by users who are confident
+      in writing confluent sets of class instances.
+
+      Includes the following: ``DeepSubsumption``, ``IncoherentInstances`` (but without implying
+      ``INCOHERENT`` on every instance), ``NoMonomorphismRestriction``, ``-Wno-warn-orphans``
+
+  Here is the for these columns:
```

Comment:
I was reading quickly and didn't realise that "these columns" refers only to some columns...

```suggestion
  Here is the key for the "semantic bundle" columns:
```